dali/pipeline/operators/reader/loader/lmdb.h

```diff
@@ -107,64 +189,50 @@ class LMDBLoader : public Loader<CPUBackend, Tensor<CPUBackend>> {
     }
 
     tensor.SetMeta(meta);
-    tensor.Resize({static_cast<Index>(value_.mv_size)});
+    tensor.Resize({static_cast<Index>(value.mv_size)});
     std::memcpy(tensor.raw_mutable_data(),
-                reinterpret_cast<uint8_t*>(value_.mv_data),
-                value_.mv_size*sizeof(uint8_t));
+                reinterpret_cast<uint8_t*>(value.mv_data),
+                value.mv_size * sizeof(uint8_t));
   }
 
  protected:
-  Index SizeImpl() override {
-    return lmdb_size_;
-  }
+  Index SizeImpl() override { return size_array_[db_path_.size()]; }
 
   void PrepareMetadataImpl() override {
-    // Create the db environment, open the passed DB
-    CHECK_LMDB(mdb_env_create(&mdb_env_), db_path_);
-    auto mdb_flags = MDB_RDONLY | MDB_NOTLS | MDB_NOLOCK;
-    CHECK_LMDB(mdb_env_open(mdb_env_, db_path_.c_str(), mdb_flags, 0664), db_path_);
-
-    // Create transaction and cursor
-    CHECK_LMDB(mdb_txn_begin(mdb_env_, NULL, MDB_RDONLY, &mdb_transaction_), db_path_);
-    CHECK_LMDB(mdb_dbi_open(mdb_transaction_, NULL, 0, &mdb_dbi_), db_path_);
-    CHECK_LMDB(mdb_cursor_open(mdb_transaction_, mdb_dbi_, &mdb_cursor_), db_path_);
-    lmdb_size_ = lmdb::LMDB_size(mdb_transaction_, mdb_dbi_, db_path_);
-
-    // Optional: debug printing
-    lmdb::PrintLMDBStats(mdb_transaction_, mdb_dbi_, db_path_);
-
+    size_array_.resize(db_path_.size() + 1);
+    size_array_[0] = 0;
+    mdb_.resize(db_path_.size());
+    for (size_t i = 0; i < db_path_.size(); i++) {
```

Comment:
shouldn't this be
```suggestion
    for (size_t i = 0; i < db_path_.size() - 1; i++) {
```
Otherwise you'll go out of bounds when you access `size_array_[i+1]`

You can also rephrase it as
```
for (size_t i = 1; i < db_path_.size(); i++) {
    size_array_[i] = size_array_[i-1] + mdb_[i-1].GetSize();
}
```

Reply:
@jantonguirao Thank you very much for your review. The member `size_array_` has` N+1` entries while `N` is the number of LMDB files. for example, the LMDB files and their entries are as follows:
lmdb0 7
lmdb1 9
lmdb2 8
then `size_array_` is `{0, 7, 16, 24}`, the first entry is always zero for convenient, the last entry is the total number of entries.